src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs

```diff
@@ -33,6 +33,10 @@ public void Execute(IExecutionContext context, Command command)
         {
             ArgUtil.NotNull(context, nameof(context));
             ArgUtil.NotNull(command, nameof(command));
+
+            context.Variables.TryGetValue(Constants.Variables.Task.PublishTelemetry, out string publishTelemetryVar);
+            if (bool.TryParse(publishTelemetryVar, out bool publishTelemetry) && !publishTelemetry) return;
```

Comment:
nipick: `return` statement should be on the new line in {} brackets
ref: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#layout-conventions